website/src/components/SearchPage/fields/DateRangeField.tsx

```diff
@@ -0,0 +1,134 @@
+    )[0];
+    const lowerToField = field.groupedFields.filter(
+        (f) => f.name.endsWith('To') && f.rangeOverlapSearch!.bound === 'lower',
+    )[0];
+    const upperFromField = field.groupedFields.filter(
+        (f) => f.name.endsWith('From') && f.rangeOverlapSearch!.bound === 'upper',
+    )[0];
+    const upperToField = field.groupedFields.filter(
+        (f) => f.name.endsWith('To') && f.rangeOverlapSearch!.bound === 'upper',
+    )[0];
+    // TODO maybe all these '!' should be handled differently?
+
+    const [strictMode, setStrictMode] = useState(
+        isStrictMode(
+            lowerFromField.name in fieldValues,
+            lowerToField.name in fieldValues,
+            upperFromField.name in fieldValues,
+            upperToField.name in fieldValues,
+        ),
+    );
+
+    const lowerField = strictMode ? lowerFromField : upperFromField;
+    const upperField = strictMode ? upperToField : lowerToField;
+
+    const [lowerValue, setLowerValue] = useState(fieldValues[lowerField.name] ?? '');
+    const [upperValue, setUpperValue] = useState(fieldValues[upperField.name] ?? '');
+
+    useEffect(() => {
+        if (strictMode) {
+            setSomeFieldValues({
+                [lowerFromField.name]: lowerValue,
+                [upperToField.name]: upperValue,
+                [upperFromField.name]: null,
+                [lowerToField.name]: null,
+            });
+        } else {
+            setSomeFieldValues({
+                [upperFromField.name]: lowerValue,
+                [lowerToField.name]: upperValue,
+                [lowerFromField.name]: null,
+                [upperToField.name]: null,
+            });
+        }
+    }, [
+        strictMode,
+        lowerValue,
+        upperValue,
+        lowerFromField,
+        lowerToField,
+        upperFromField,
+        upperToField,
+        setSomeFieldValues,
+    ]);
+
+    return (
+        <div key={field.name} className='flex flex-col border p-3 mb-3 rounded-md border-gray-300'>
+            <div className='flex flex-row justify-between items-baseline mb-2'>
+                <h3 className='text-gray-500 text-sm'>{field.displayName}</h3>
+                <CustomTooltip id={'strict-tooltip' + field.name}>
+                    <div className='w-52'>{TOOLTIP_TEXT}</div>
+                </CustomTooltip>
+                <label data-tooltip-id={'strict-tooltip' + field.name}>
+                    <span className='text-gray-400 text-sm mr-2'>strict</span>
+                    <input
+                        type='checkbox'
+                        className='checkbox checkbox-sm [--chkbg:theme(colors.gray.400)]'
+                        checked={strictMode}
+                        onChange={(event) => setStrictMode(event.target.checked)}
+                    />
+                </label>
+            </div>
+
+            <DateField
+                field={{
+                    name: 'not used',
+                    label: 'Lower',
```

Comment:
IMO `From` and `To` is significantly more understandable

Reply:
(to be clear, for code the current is fine, and arguably better given we use `From` for the individual LAPIS query - just talking in terms of UI)

Reply:
https://github.com/loculus-project/loculus/pull/3308/commits/1daa04f01bd7f9b8f2db084d6e18a4204a85accf